### PR TITLE
Update repo to use GH actions

### PR DESCRIPTION
**Pull request by eiri**
## PR summary

This is a test to evaluate the merit of switching from Travis to github actions for simple version matrix tests

## PR Checklist

Please make sure that your PR fulfills the following requirements:

- [x] The commit message follows the
[Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type  

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [x] Build/CI related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      with:
+        go-version: ${{ matrix.go }}
+    - name: Checkout code
+      uses: actions/checkout@v3
+    - name: Run tests
+      run: go test ./auth/... ./common/... ./cloudantv1/... -race -coverprofile=coverage.txt -covermode=atomic
+    - name: Lint
+      uses: golangci/golangci-lint-action@v3
+      with:
+        version: v1.45.1
```
Comment (ricellis) [inline .github/workflows/test.yml:29]: I believe the default is latest, any reason we can't use that and need this specific version?
```suggestion
```

**Review by ricellis — CHANGES_REQUESTED** [review]
I think the removal of travis should be independent of this change. I'll remove the travis merge requirement and we can evaluate them in parallel and see if the actions org limits present us problems.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+name: test
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
+env:
+  GO111MODULE: on
```
[the remark this answers is not in the document]
Comment (eiri) [reply .github/workflows/test.yml:10]: Don't think we need it, just a carry over from travis

File: .github/workflows/test.yml (hunk 1 above)
Comment (eiri) [reply .github/workflows/test.yml:29]: It's pined in travis, so I mimic it here, thought it has some significance.

**Review by ricellis — APPROVED** [review]

**Review by vmatyusGitHub — APPROVED** [review]

**Pull request merged**